### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            _requests[1].IsMatch = true;
+            _requests[1].Values = new RouteValueDictionary(new { id = 17 });
+
+            _requests[2].HttpContext = new DefaultHttpContext();
+            _requests[2].HttpContext.Request.Path = "/api/Widgets/search/dldldldldld/ddld";
+            _requests[2].IsMatch = false;
+            _requests[2].Values = new RouteValueDictionary();
+        }
+
+        [Benchmark(Description = "Attribute Routing", OperationsPerInvoke = 100 * 3)]
```
Comment (benaadams) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: Const the values?

File: test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        public async Task AttributeRouting()
+        {
+            for (var i = 0; i < 100; i++)
```
Comment (benaadams) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: const value

File: test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+            treeBuilder.MapInbound(handler, TemplateParser.Parse("api/Widgets"), "default", 0);
+            treeBuilder.MapInbound(handler, TemplateParser.Parse("api/Widgets/{id}"), "default", 0);
+            treeBuilder.MapInbound(handler, TemplateParser.Parse("api/Widgets/search/{term}"), "default", 0);
+            treeBuilder.MapInbound(handler, TemplateParser.Parse("admin/users/{id}"), "default", 0);
+            treeBuilder.MapInbound(handler, TemplateParser.Parse("admin/users/{id}/manage"), "default", 0);
+
+            _treeRouter = treeBuilder.Build();
+
+            _requests = new RequestEntry[3];
```
Comment (benaadams) [inline test/Microsoft.AspNetCore.Routing.Performance/RoutingBenchmark.cs]: const value

**Comment by rynowak** [comment]
6c79e84d1b6b45b6f94ada27ff333ab4112fb881 ca66e89899f5d09cd28fa2e071dc2f7d12f356fe

**Pull request closed without merging**